Python/Product/PythonTools/PythonTools/Navigation/WaitForCompleteAnalysisDialog.xaml.cs

```diff
@@ -46,21 +46,27 @@ partial class WaitForCompleteAnalysisDialog : DialogWindowVersioningWorkaround {
 
         private bool UpdateItemsRemaining(int itemsLeft) {
             if (itemsLeft == 0) {
-                Dispatcher.Invoke((Action)(() => {
-                    this.DialogResult = true;
-                    this.Close();
-                }));
+                try {
+                    Dispatcher.Invoke((Action)(() => {
+                        this.DialogResult = true;
+                        this.Close();
+                    }));
+                } catch (OperationCanceledException) {
+                    // Should only occur if the dialog is closed already, so nothing left to do
```

Comment:
Do you have a call stack for this exception?

Reply:
In #3720, but it's not clear which one has the problem. But AIUI the only way the dispatcher starts cancelling pending operations is if the dispatcher is gone, which means the window is gone.